src/paho/mqtt/client.py

```diff
@@ -2671,6 +2694,89 @@ def _reconnect_wait(self):
             time.sleep(min(remaining, 1))
             remaining = target_time - time_func()
 
+    def _get_proxies(self):
+        def proxy_is_valid(p):
+            if isinstance(p, dict):
+                return bool(p.get("proxy_type"))
+            elif isinstance(p, (list, tuple)):
+                return len(p) == 6 and bool(p[0])
+            else:
+                return False
+
+        proxies = []  # proxy servers to use in order of preference
+
+        if proxy_is_valid(self._proxy):
+            proxies.append(self._proxy)
```

Comment:
If the user *explicitly* give a proxy parameter to the connect() method, shouldn't we only try this proxy ?

In general shouldn't we only try the first proxy ? We could also wonder why using environment variable if user explicitly called socks.set_default_proxy().